### PR TITLE
Mainmenu: Fix problems with recently added "game" type

### PR DESCRIPTION
**Pull request by SmallJoker**
Fixes problems pointed out in http://irc.minetest.ru/minetest-dev/2017-05-18#i_4927850
Caused by #5367

### CODE REVIEW

File: src/client.cpp [hunk 1]
```diff
@@ -168,7 +168,7 @@ void Client::initMods()
 		if (!string_allowed(mod.name, MODNAME_ALLOWED_CHARS)) {
 			throw ModError("Error loading mod \"" + mod.name +
 				"\": Mod name does not follow naming conventions: "
-					"Only chararacters [a-z0-9_] are allowed.");
+					"Only characterss [a-z0-9_] are allowed.");
```
Comment (kilbith) [inline src/client.cpp]: typo

File: builtin/mainmenu/modmgr.lua [hunk 2]
```diff
@@ -252,7 +252,7 @@ function modmgr.render_modlist(render_list)
 					break
 				end
 			end
-		elseif v.typ == "game_mod" or v.typ == "game" then
+		elseif string.find(v.typ, "game") then
```
Comment (kilbith) [inline builtin/mainmenu/modmgr.lua]: You really need `find` here and there? Prefer `sub` when it's possible.

**Comment by SmallJoker** [comment]
@nerzhul Instead of creating a function for this simple task, I extended the mod table to mark all the game content with the key/value `is_game_content = true`, which is IMO a better solution for this check.

**Comment by nerzhul** [comment]
travis failed

**Pull request merged**